packages/react-devtools-shared/src/devtools/views/Profiler/SidebarSelectedFiberInfo.js

```diff
@@ -207,7 +207,7 @@ function WhatChanged({
   if (changes.length === 0) {
     changes.push(
       <div key="nothing" className={styles.WhatChangedItem}>
-        The parent component rendered.
+        The parent component rendered or context changed
```

Comment:
FWIW, we _can_ detect a context change for class components, so we probably only want to show this additional warning if we're inspecting a function component.

Reply:
Sure, how do I detect that situation?

BTW, can't we also detect a context change for function components, simply by the process of elimination? If it's not due to state changes and it's memoized, then it must be due to context changes, right?

Reply:
> Sure, how do I detect that situation?

You can use the `type` attribute to differentiate between a class component (`ElementTypeClass`) and a function component (`ElementTypeFunction`, `ElementTypeMemo`, or `ElementTypeForwardRef`).

> BTW, can't we also detect a context change for function components, simply by the process of elimination? If it's not due to state or prop changes and it's memoized, then it must be due to context changes, right?

I think that may be a shaky assumption.

Currently the only two hooks that _directly_ trigger a re-render are `useReducer` and `useState` (which is built on top of `useReducer`). The rest of them are either _reactive_ or at least _indirect_ (like `useContext`). I'm not sure if this will always be the case though.

So I guess we _could_ show a more meaningful message if memoized props and state were shallowly equal _and_ element `type` was `ElementTypeMemo` but this feels fragile and wouldn't cover some more ambiguous cases. If we show a more specific message in some cases, but not in others- that alone could cause confusion. ("It's not a context change, because if it would DevTools would say so like it did before.")